src/core/storage-services/Holochain/HolochainService.ts

```diff
@@ -230,6 +307,32 @@ export default class HolochainService {
             } catch(e) {
                 console.error("HolochainService: ERROR activating app", lang, " - ", e)
             }
+        } else {
+            for (let dna of dnas) {
+                if (callback != undefined) {
+                    console.log("HolochainService: setting holochains signal callback for language", lang);
+                    let infoResult = await this.#appWebsocket!.appInfo({installed_app_id: lang})
+                    const { cell_data } = infoResult
+                    const cell = cell_data.find(c => c.cell_nick === dna.nick)
+                    if(!cell) {
+                        const e = new Error(`No DNA with nick '${dna.nick}' found for language ${lang}`)
+                        console.error(e)
+                        return e
+                    }
+                    const hash = cell.cell_id[0];
+                    
+                    const hashHex = hash.toString("hex");
+                    let callbacks = this.signalCallbacks.get(hashHex);
+                    let newCallbacks = [];
+                    if (callbacks) {
+                        callbacks.push(callback);
+                        newCallbacks = callbacks;
+                    } else {
+                        newCallbacks = [callback] as AppSignalCb[]
+                    }
+                    this.signalCallbacks.set(hashHex, newCallbacks);
```

Comment:
The signal callback registration can be extracted out of the two cases to avoid c/p code. See my PR: https://github.com/perspect3vism/ad4m-executor/pull/29/files#diff-7cb52e8044275b999765d1fd40e5b6810eb9d1a0fbf0781dc523dc77810bb994

Reply:
Ill just merge your code here